src/main/java/org/springframework/integration/kafka/listener/QueueingMessageListenerInvoker.java

```diff
@@ -78,14 +95,8 @@ public void enqueue(KafkaMessage message) {
 			// handle the case when the thread is interrupted while the adapter is still running
 			// retry adding the message to the queue until either we succeed, or the adapter is stopped
 			while (!added && this.running) {
-				try {
-					this.messages.put(message);
-					added = true;
-				}
-				catch (InterruptedException e) {
-					// we ignore the interruption signal if we are still running, but pass it on if we are stopped
-					wasInterruptedWhileRunning = true;
-				}
+				ringBufferProcessor.onNext(message);
+				added = true;
```

Comment:
Actually, this is wrong. The thread calling onNext(message) will block if the buffer is full, therefore the previous way of handling interruptions must be reinstated. Will issue a PR.